library/src/lib/calendar/calendar2/calendar2.component.ts

```diff
@@ -52,7 +56,17 @@ export type DaysOfWeek = 1 | 2 | 3 | 4 | 5 | 6 | 7;
         '[attr.id]': 'id'
     }
 })
-export class Calendar2Component implements OnInit, ControlValueAccessor {
+export class Calendar2Component implements OnInit, OnDestroy, ControlValueAccessor {
+
+    // --------------- CODE CONCERNING THE MONTH VIEW --------------- 
+    /** @hidden For i18n, the list of month names. Internal use. */
+    monthNames: string[];
```

Comment:
You should get these directly from the service and not need an intermediary variable

Reply:
Just followed the Architecture.

Reply:
should be discussed then, not sure why the monthNames is a part of that if you can just get it directly